main.go

```diff
@@ -113,11 +116,19 @@ func main() {
 		vpnResolver.SetPermanentDNS(customDns.HostName, ipAddr)
 		log.Printf("Add custom DNS: %s -> %s\n", customDns.HostName, customDns.IP)
 	}
+	localResolver := service.NewDnsServer(vpnResolver, []string{conf.ZJUDNSServer, conf.SecondaryDNSServer})
+	vpnStack.SetupResolve(localResolver)
+
+	go vpnStack.Run()
 
 	vpnDialer := dial.NewDialer(vpnStack, vpnResolver, ipResource, conf.ProxyAll)
 
 	if conf.DNSServerBind != "" {
-		go service.ServeDNS(conf.DNSServerBind, vpnResolver)
+		go service.ServeDNS(conf.DNSServerBind, localResolver)
+	}
+	if conf.TUNMode {
```

Comment:
这里考虑增加 conf.DNSHijack 这个条件吗

```suggestion
	if conf.TUNMode && conf.DNSHijack {
```

Reply:
感觉没问题 😄

Reply:
okk，仔细看了一下确实